controllers/nonadmin_controller.go

```diff
@@ -86,6 +85,21 @@ func (r *DPAReconciler) ReconcileNonAdminController(log logr.Logger) (bool, erro
 		return true, nil
 	}
 
+	selector, err := fields.ParseSelector(fmt.Sprintf("metadata.namespace!=%s", r.NamespacedName.Namespace))
+	if err != nil {
+		return false, err
+	}
+	dpaList := &oadpv1alpha1.DataProtectionApplicationList{}
+	err = r.ClusterWideClient.List(r.Context, dpaList, &client.ListOptions{FieldSelector: selector})
+	if err != nil {
+		return false, err
+	}
+	for _, dpa := range dpaList.Items {
+		if (&DPAReconciler{dpa: &dpa}).checkNonAdminEnabled() {
+			return false, fmt.Errorf("only a single instance of Non-Admin Controller can be installed across the entire cluster. Non-Admin controller is also configured to be installed in %s namespace", dpa.Namespace)
```

Comment:
Lets re-frame the error text a little bit: `only a single instance of Non-Admin Controller can be installed across the entire cluster. Non-Admin controller is already configured and installed in %s namespace", dpa.Namespace`

Reply:
I would like to keep as it is

from what I tested, NAC will never be deployed 2 or more times. Either one is installed (one was installed and now a second is trying to be installed) or none (2 or more are trying to be installed at the same time)

So, your suggestion would make sense to the case of the DPA that tried to install NAC and it already exists in cluster, but not for the DPA that already installed it, or other cases (error messages will appear in all DPAs, because DPA reconciles every minute). Do you agree?

Reply:
Users does the following:
- Creates DAP 1
- Enables NAC for DPA 1
- Now Another NS, OADP is installed and DPA 2 is created there 
- User enabled NAC for DPA 2
I want error message only on DPA 2.

Reply:
To have error only on "wrong" DPAs, I would need to know if NAC is already deployed (do a GET call for deployments in OADP namespace) prior to erroring out. Would this not complicate validation too much? (and every minute, DPA would do GET call)

Reply:
DPA 1 did nothing wrong, why should DPA 1 be errored out ?

Reply:
Just because we can not be sure that error is not in DPA 1

examples

- DPA 1 is created first with NAC enabled -> DPA 2 created after with NAC enabled -> by doing a deployment get call we can add error only to DPA 2 ✅

- DPA 1 and DPA 2 are created (or edited) at the same time with NAC enabled -> in this case, which DPA we error out? ❌

Reply:
For case 2, we don't decide that, let the DAP controller do its work. Either of the DPAs will get the error for sure, right ?

Reply:
@shubham-pampattiwar that is my fear. I think if your suggest approach, both will deploy NAC

@mpryc NAC has leader election, I will test what happens